### DISCUSSION TITLE
[SUGGESTION] Improved/ more detailed fuel tank hitboxes

### DISCUSSION THREAD

**Discussion by TIKIRobo** [tsunamayo/Starship-EVO#4038] [Suggestions]
Currently the new fuel blocks have an awkward hitbox 
![image](https://user-images.githubusercontent.com/56371294/110490811-e7b49080-80ad-11eb-94f5-ddf0a7cf7183.png)
Here im trying to make a more "form fitting" casing around the fuel tanks but due to their rectangular hitboxes this is not possible 
With a more accurate hitbox it would be possible

**Comment by ProPeach** [comment]
Working these tanks into a blocky interior is a little tough, perhaps the tanks could have a frame already in their model to make this easier?
![image](https://user-images.githubusercontent.com/27857711/110504487-1d309c80-80f5-11eb-9952-16cd1f0e7e30.png)
This is for the 3m tank, the grey frame uses shapes and geometries that are compatible with bricks (for example, 0.25x0.5m slopes), so that players can integrate them easily into a ship.

**Comment by TIKIRobo** [comment]
that could also work as then the ribs that im trying to add would look like they had an attachment point

**Comment by Kaiser-Indrasil** [comment]
@tsunamayo I was thinking that the model looks a bit jagged and low poly and thought to myself, "why not make it even more edgy instead?" I was inspired by the spaceship Algos from Eve Online which has these cool fuel tank-looking shapes:

![image](https://user-images.githubusercontent.com/63756057/110506274-d8a60080-80f6-11eb-9587-fc817db1b1d7.png)

However, these would not work well with the pipe connectors and the voxel nature of the game, **and** stretch tech on top of all this.
So I started digging, and I dug up this: **a truncated cuboctahedron**.
I came up with a model that features dark accents which are prevalent accross different systems in Starship Evo and scales nicely.
The blue sections are recolorable.
The bevelled edges are 0.5m x 0.5m for the block tank and 12.5cm x 12.5cm for the brick tank.

![ssdgfdg](https://user-images.githubusercontent.com/63756057/110530477-680cdd00-8112-11eb-9fcd-17739296cb2a.jpg)
![SketchUp_GdEjMjHSOA](https://user-images.githubusercontent.com/63756057/110530564-85da4200-8112-11eb-968e-a63db1604300.png)
![SketchUp_dIth2IJfhn](https://user-images.githubusercontent.com/63756057/110530593-8c68b980-8112-11eb-84f5-c3dd21bb598d.png)

**Comment by TIKIRobo** [comment]
@Kaiser-Indrasil that type of fuel tank looks super cool but it seems like any sort of "ribs" that could be wrapped around it would have a small gap but maybe im just dumb

**Comment by tsunamayo** [comment]
Yes the hitbox is still wip. I am currently refactoring the collision engine so it might have to wait a bit.
@ProPeach @Kaiser-Indrasil you two should make me some official brick some day. You seem better at that than I am! I was thinking of the first option but opt out of it, I prefer the sleek look (but of course builder can always customize on top of).
Kaiser looks sleek, maybe for something more advanced? I was thinking of maybe having weapon use plasma, that would require power to be stored.

**Comment by ProPeach** [comment]
Indrasil's is definitely the more "High tech" looking option, maybe it'd be better for some fancier fuels for warp or higher quality regular fuel

**Comment by Womble-TC** [comment]
Not sure how helpful this is, but I made some fuel tank concepts a while ago. I used some existing parts from textures to make them look real. (On a side note, I'd be interested to learn how you make models and textures, since a lot of the textures in-game appear to have standerdised details and stuff): 

![tank 2](https://user-images.githubusercontent.com/29961381/110561479-a8cf1b00-813f-11eb-8f1d-01523255f1fd.png)
![tank 1](https://user-images.githubusercontent.com/29961381/110561485-aa98de80-813f-11eb-91b8-f39ab81cf8e9.png)

**Comment by Kaiser-Indrasil** [comment]
@tsunamayo Thanks, I tried to go with the design language that you and GrayShuko established. I'm studying 3D modelling and animation so I might come in handy one day ;)

**Comment by Garrett-C** [comment]
@Womble-TC That design looks like it could be cool for some sort of addon for reactors or something. Maybe if reactors ever get like a small tank addon something like that could be a cool look for it.

**Comment by Xerma** [comment]
I could see it as a sort of fuel injector for reactors to consume fuel faster in order to make more power. Almost like how slapping on a supercharger works on an irl engine

**Comment by tsunamayo** [comment]
@Womble-TC so I wanted something pretty similar for the potential plasma holder I had in mind... It would require power to store, and plasma is feeding laser / beam. I need more thinking about this, but I want to refactor the side addon and tie to game progression.

**Comment by tsunamayo** [comment]
@ProPeach @Womble-TC @Kaiser-Indrasil Hey guys, as you have some strong (at least as good as fine is appears) 3d skills - would you like to try to create some asset for the game?

**Reply by ProPeach** [reply]
I'd love to try! What kinds of models do you need? And in what format etc

**Reply by Womble-TC** [reply]
Sure I can give it a go!

**Reply by Kaiser-Indrasil** [reply]
Absolutely! Like Peach has already stated, we need a file type, and some info on the textures, poly count etc.

**Reply by Womble-TC** [reply]
Also yeah, for textures, we'd need to know what sort of material workflow you have, since every model has a similar base material, is paintable/stretchable, etc

**Reply by Xerma** [reply]
Would like to give it a shot as well of that's fine with you sir tsuna. Now that my classes are entirely finished I have plentiful free time

**Reply by tsunamayo** [reply]
@ProPeach @Womble-TC @Xerma @Kaiser-Indrasil You guys should have my mail? Could one of you drop me a mail with everybody in copy? I will write a doc with all you need to know. The hard part is the texturing, it is quite complex with a lot of caveheat, especially for bricks that rescale... I guess the issue for you would be to try the brick in game... I will see what I can do in that regard.

**Reply by Xerma** [reply]
Yeah we should still have your email, I'll message everyone on discord about it. As for testing, would it be doable to have a private build with low end modding? Or would that be more time consuming for you?

**Reply by Horkas** [reply]
@tsunamayo Would you be open to the idea of me creating some 3D enemy concepts and rigging them up/giving them some basic animations for something for players to fight in the upcoming survival update? (Or even just as test dummies/place holders for internal testing.)

All I would need to know is the Vert. count you would want from them as a guideline for me to stay within. (I'm estimating 5-15k pollys depending on enemy size.)

My idea for the concept in my head, is to make a set of stylized robots that are part of a "Mega-Corp" faction, that have been sent to planets to mine them and guard the corporations claim. So if a player lands on a corpo claimed planet and starts to mine its resources or build on it, the robots would start attacking you with a wanted timer/wave assault until the faction hq is destroyed on the planet. (Very similar to Zirax in Empyrion Galactic Survival, or the drones from No mans sky. Its a very solid enemy concept!)

This would help fit the enemies onto pretty much any planet you could want, and I can design them to be highly modular. As in: I make the base mesh robot, and add separate armors and weapons to them to make them look varied. This will ease the production time, while still letting allot of variety take place.

I can draw up the ideas of how the robots would look, and if you think they would fit the games theme, I can then start modeling away!

I have nothing but free time, and I really really want to help out this project! Anything I make would be donated to you for free, obviously :D

**Reply by TIKIRobo** [reply]
@Horkas you should probably make a separate post about that, instead of commenting on a post from 2 years ago

**Reply by Horkas** [reply]
Hah yeah, I was up super late when I wrote this, and didn't notice the time stamps. I'll make a separate post.
